### PR TITLE
Clay Date Picker

### PR DESCRIPTION
**Pull request by pat270**
Hey @matuzalemsteles,

I ended up changing most of the markup. Sorry! I decided to use `button`s instead of `div`s so we don't need to add `tabindex` to everything, only the `active` button. Also we have access to the `disabled` attribute that way disabled items don't focus on click.

I moved the `dropdown-menu` outside of the `input-group` that way we can make the input smaller without the `dropdown-menu` becoming smaller.

I wrapped a lot of stuff in containers, mostly to make the calendar collapse in smaller screens gracefully. You can see the markup in the last commit and also on the Test Site under Form Elements (Date Picker). Let me know if there are any issues with this.

### CODE REVIEW

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: packages/clay-css/src/scss/components/_date-picker.scss [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	display: flex;
+	padding-left: 0.25rem;
+	padding-right: 0.25rem;
+}
+
+.date-picker-nav-item-expand {
+	flex-grow: 1;
+}
+
+.date-picker-nav-controls {
```
Comment (matuzalemsteles) [inline packages/clay-css/src/scss/components/_date-picker.scss:42]: I think the `date-picker-nav-controls` buttons should be grays, right?

Comment (matuzalemsteles) [reply packages/clay-css/src/scss/components/_date-picker.scss:42]: From what Lexicon has in Figma the buttons are more spaced with 8px between them.

**Review by matuzalemsteles — COMMENTED** [review]
hey @pat270 Great Job!! 👏🏻

File: packages/clay-css/src/scss/components/_date-picker.scss [hunk 2]
```diff
@@ -0,0 +1,103 @@
+.date-picker {
+	.input-group-text {
+		@include clay-input-group-text-variant($date-picker-input-group-text);
```
Comment (matuzalemsteles) [inline packages/clay-css/src/scss/components/_date-picker.scss:3]: hey @pat270, I think we'll need this customization also for ClayTimePicker outside of DatePicker as well.

Comment (pat270) [reply packages/clay-css/src/scss/components/_date-picker.scss:3]: @matuzalemsteles we can add it when we make ClayTimePicker. I'd like to keep them separate even though Lexicon defines the styles the same.

Comment (matuzalemsteles) [reply packages/clay-css/src/scss/components/_date-picker.scss:3]: ok, np.

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**